bazel/defs.bzl

```diff
@@ -10,7 +10,77 @@
       (e.g. switch platforms)
     HOST_ARCHES is the set of support host architectures
       (e.g. for running against a simulator or testing)
+    STRATUM_DEFAULT_COPTS are the default flags passed to the compiler.
 """
-EMBEDDED_ARCHES = [ "x86" ]
-HOST_ARCHES = [ "x86" ]
-STRATUM_INTERNAL = [ "//stratum:__subpackages__" ]
\ No newline at end of file
+
+load(
+    "@com_google_absl//absl:copts/GENERATED_copts.bzl",
+    "ABSL_GCC_FLAGS",
+    "ABSL_GCC_TEST_FLAGS",
+    "ABSL_LLVM_FLAGS",
+    "ABSL_LLVM_TEST_FLAGS",
+)
+
+EMBEDDED_ARCHES = ["x86"]
+HOST_ARCHES = ["x86"]
+STRATUM_INTERNAL = ["//stratum:__subpackages__"]
+
+# Compiler warnings for different toolchains.
+STRATUM_COMPILER_WARNINGS_COMMON = [
+    "-Wall",
+]
+STRATUM_DISABLED_COMPILER_WARNINGS_COMMON = [
+    "-Wno-sign-compare",
+    "-Wno-unused-function",
+    "-Wno-unused-parameter",
+    "-Wno-unused-variable",
+]
+STRATUM_DISABLED_COMPILER_WARNINGS_GCC = []
+STRATUM_DISABLED_COMPILER_WARNINGS_LLVM = [
+    "-Wno-gnu-zero-variadic-macro-arguments",
+    "-Wno-missing-variable-declarations",
+    "-Wno-suggest-destructor-override",
+    "-Wno-unused-lambda-capture",
+    "-Wno-unused-template",
+]
+STRATUM_DISABLED_COMPILER_WARNINGS = STRATUM_DISABLED_COMPILER_WARNINGS_COMMON + select({
+    "//stratum:llvm_compiler": STRATUM_DISABLED_COMPILER_WARNINGS_LLVM,
+    "//conditions:default": STRATUM_DISABLED_COMPILER_WARNINGS_GCC,
+})
+
+# Compiler warnings that are threated as errors.
+STRATUM_COMPILER_ERRORS_COMMON = [
+    "-Werror=ignored-attributes",
+    "-Werror=return-type",
+]
+STRATUM_COMPILER_ERRORS_GCC = [
+    "-fdiagnostics-color=always",
+    "-Werror=reorder",
+]
+STRATUM_COMPILER_ERRORS_LLVM = [
+    "-Werror=implicit-fallthrough",  # TODO(max): move to common after gcc 7
+    "-Werror=reorder-ctor",
+    "-Werror=thread-safety-analysis",
+]
+STRATUM_COMPILER_ERRORS = STRATUM_COMPILER_ERRORS_COMMON + select({
+    "//stratum:llvm_compiler": STRATUM_COMPILER_ERRORS_LLVM,
+    "//conditions:default": STRATUM_COMPILER_ERRORS_GCC,
```

Comment:
Match on gcc, empty default

Reply:
GCC is not detected by Bazel: https://github.com/bazelbuild/bazel/issues/12707

But using gcc by default seems common practice and safe.